core/src/main/java/com/homeaway/streamplatform/streamregistry/db/dao/impl/StreamDaoImpl.java

```diff
@@ -58,18 +61,41 @@
 
     private StreamValidator streamValidator;
     private SchemaManager schemaManager;
+    private KafkaManager kafkaManager;
+
+    @Inject
+    public StreamDaoImpl(ManagedKafkaProducer managedKafkaProducer,
+        ManagedKStreams kStreams,
+        @Named("stream-registry-env") String env,
+        RegionDao regionDao,
+        ManagedInfraManager managedInfraManager,
+        StreamValidator validator,
+        SchemaManager schemaManager) {
+        this(managedKafkaProducer, kStreams, env, regionDao, managedInfraManager.getInfraManager(), validator, schemaManager);
+    }
 
     public StreamDaoImpl(ManagedKafkaProducer managedKafkaProducer,
                          ManagedKStreams kStreams,
                          String env,
                          RegionDao regionDao,
                          InfraManager infraManager,
-                         KafkaManager kafkaManager,
                          StreamValidator validator,
                          SchemaManager schemaManager) {
-        super(managedKafkaProducer, kStreams, env, regionDao, infraManager, kafkaManager);
+        this(managedKafkaProducer, kStreams, env, regionDao, infraManager, validator, schemaManager, new KafkaManagerImpl());
```

Comment:
Remove unnecessary public ctor if not needed.

Reply:
Same as your previous observation - its been called from the IT flow ( [here](https://github.com/homeaway/stream-registry/blob/master/core/src/test/java/com/homeaway/streamplatform/streamregistry/resource/BaseResourceIT.java#L246-L247) )

( Maybe we've an opportunity here to streamline the IT implementation here. Will try to see if we can do it in this PR or shall we have a follow up )

Reply:
Same as your previous observation - its been called from the IT flow ( [here](https://github.com/homeaway/stream-registry/blob/master/core/src/test/java/com/homeaway/streamplatform/streamregistry/resource/BaseResourceIT.java#L246-L247) )

( Maybe we've an opportunity here to streamline the IT implementation here. Will try to see if we can do it in this PR or shall we have a follow up )

Reply:
( also we've moved the KafkaManager inside `StreamDaoImpl` )

Reply:
Changed this flow.